pandas/plotting/_core.py

```diff
@@ -875,6 +886,71 @@ def _make_plot(self):
             ax.errorbar(data[x].values, data[y].values,
                         linestyle='none', **err_kwds)
 
+    def _sci_notation(self, num):
+        '''
+        Returns mantissa and exponent of the number passed in agument.
+        Example:
+        _sci_notation(89278.8924)
+        >>> (8.9, 5.0)
+        '''
+        scientific_notation = '{:e}'.format(num)
+        expnt = float(re.search(r'e([+-]\d*)$',
+                                scientific_notation).groups()[0])
+        coef = float(re.search(r'^([+-]?\d\.\d)',
+                               scientific_notation).groups()[0])
+        return coef, expnt
+
+    def _legend_bubbles(self, s_data_max, size_factor, bubble_points):
+        '''
+        Computes and returns appropriate bubble sizes and labels for the
+        legend of a  bubble plot. Creates 4 bubbles with round values for the
+        labels, the largest of which is close to the maximum of the data.
+        '''
+        coef, expnt = self._sci_notation(s_data_max)
+        labels_catalog = {
+            (9, 10): [10, 5, 2.5, 1],
+            (7, 9): [8, 4, 2, 0.5],
+            (5.5, 7): [6, 3, 1.5, 0.5],
+            (4.5, 5.5): [5, 2, 1, 0.2],
+            (3.5, 4.5): [4, 2, 1, 0.2],
+            (2.5, 3.5): [3, 1, 0.5, 0.2],
+            (1.5, 2.5): [2, 1, 0.5, 0.2],
+            (0, 1.5): [1, 0.5, 0.25, 0.1]
+        }
+        for lower_bound, upper_bound in labels_catalog:
+            if (coef >= lower_bound) & (coef < upper_bound):
+                labels = 10**expnt * np.array(labels_catalog[lower_bound,
+                                                             upper_bound])
+                sizes = list(bubble_points * size_factor * labels / s_data_max)
+                labels = ['{:g}'.format(l) for l in labels]
+                return (sizes, labels)
+
+    def _make_legend(self):
+        ax = self.axes[0]
+        if hasattr(self, "size_title"):
+            size_title = self.size_title
+            s_data_max = self.s_data_max
+            size_factor = self.size_factor
+            bubble_points = self.bubble_points
+            import matplotlib.legend as legend
+            sizes, labels = self._legend_bubbles(s_data_max,
+                                                 size_factor,
+                                                 bubble_points)
+            bubbles = []
+            for size in sizes:
+                bubbles.append(ax.scatter([],
+                                          [],
+                                          s=size,
+                                          color='white',
```

Comment:
Should this be `color=rcParams['axes.facecolor']` and `axes.edgecolor` for the next line?

Reply:
This color should not matter as there are no data points in it.

It is probably better to just create a `Collection` here and not bother adding it to the Axes (to not clutter the draw tree with dummy artists).

See https://matplotlib.org/users/legend_guide.html#creating-artists-specifically-for-adding-to-the-legend-aka-proxy-artists

Reply:
@tacaswell Thank you for your help, I had tried this but used Circle objects instead of CircleCollection objects and could not make it work. Now it works :) And it's cleaner indeed.